src/etl/ReportingETL.cpp

```diff
@@ -208,6 +217,20 @@ ReportingETL::publishLedger(ripple::LedgerInfo const& lgrInfo)
     else
         BOOST_LOG_TRIVIAL(info) << __func__ << " - Skipping publishing ledger "
                                 << std::to_string(lgrInfo.seq);
+    if (!writing_)
+    {
+        BOOST_LOG_TRIVIAL(debug) << __func__ << " - Updating cache";
+
+        std::vector<Backend::LedgerObject> diff =
+            Backend::synchronousAndRetryOnTimeout([&](auto yield) {
+                return backend_->fetchLedgerDiff(lgrInfo.seq, yield);
+            });
+
+        backend_->cache().update(diff, lgrInfo.seq);
+        backend_->txCache().update(hashes, transactions, lgrInfo.seq);
+        backend_->updateRange(lgrInfo.seq);
+    }
```

Comment:
Let's do all this before we publish the new ledger and transactions, which is when we call `subscriptions_->pubLedger()`